src/Nuget.PackageIndex.VisualStudio/CodeFixes/PackageInstaller.cs

```diff
@@ -137,17 +137,20 @@ private void OnPackageInstalled(IVsPackageMetadata metadata)
             // sync at least one of them would do it and add all new packages to the index.
             var indexFactory = new PackageIndexFactory();
             var builder = indexFactory.GetLocalIndexBuilder();
-            builder.BuildAsync(newOnly: true);
+            builder.BuildAsync(newOnly: true, cancellationToken:CancellationToken.None);
         }
 
-        ~PackageInstaller()
+        #region IDisposable
+
+        public void Dispose()
         {
-            Dispose();
+            Dispose(true);
+            GC.SuppressFinalize(this);
         }
 
-        public void Dispose()
```

Comment:
more of a question, make sure you actually want to handle this through dispose. And test your code in shutdown scenarios.

Typically you want to use dispose only when you have to get rid of native resources. It might be the right thing in your case though I'm not sure. Just a good thing to look at before picking the pattern